specs/interop/token-bridging.md

```diff
@@ -2,96 +2,298 @@
+
+Here is what a possible implementation could look like:
+
+```solidity
+function xApprove(address _spender, uint256 _chainId, uint256 _amount) external returns (bool) {
+  _xAllowances[msg.sender][_spender][_chainId] = _amount;
+  emit XApproval(msg.sender, _spender, _chainId, _amount);
+  return true;
+}
+// this will override the ERC20 view function
+function allowance(address _owner, address _spender) external view returns (uint256) {
+  return allowance(_owner, _spender, block.chainid);
+}
+
+function allowance(address _owner, address _spender, uint256 _chainId) external view returns (uint256) {
+  if (_chainId == block.chainId) return _allowance[_owner][_spender];
+  return _xAllowance[_owner][_spender][_chainId];
 }
 ```
 
-## L1 Native Tokens
+It is technically possible for the standard to also have remote approvals (initialize in chain A and approval for funds in chain B). This feature, however, was discarded as an address might have different owners in different chains (think of smart wallets for instance). Moreover, even if this was not an issue, switching `chainId` is not that relevant UX-wise to include a dedicated function in the token standard.
+
+#### `remoteTransferFrom()`
 
-To support L1 native tokens with cross chain liquidity, the `OptimismMintableERC20Token`
-interface MUST be supported in addition to the `SuperchainERC20` interface.
+As already mentioned, the `remoteTransferFrom()` will allow using funds on a remote chain on behalf of another party. The function should initiate a call to the `L2ToL2CrossChainMessenger` without burning anything on the origin chain.
 
-## Cross Chain `transferFrom`
+To process a `remoteTransferFrom()` on the destination, the standard should include a `relayTransferFrom()` function that checks local allowance and calls `burn()`. Then, it will check if the target lives on the same `chainId` to call `relayAndExecute()` to the corresponding contract.
 
-Should an overloaded `approve` method be added to the interface so that a user can
-approve a contract on a remote domain to bridge on their behalf? A new `allowances`
-mapping could be added to the contract to enable this.
+Finally, the standard should included a `relayAndExecute()` function that might be used to resume execution on the origin chain. This is how the atomicity lost in the `transferFrom()` gets addressed.
 
-## Factory
+A possible implementation for these functionalities would look like this:
 
-A token factory predeploy can be used to ensure that `SuperchainERC20` tokens
-can be permissionlessly and deterministically deployed. If a deterministic deployment
-scheme is not used, maintaining a mapping of the token addresses between all of the
-chains will not be scalable.
+```solidity
+function remoteTransferFrom(
+  address _owner,
+  address _recipient,
```

Comment:
Is it ok if we remove this implementation/definition of the interface for now and instead just talk about it as being a future addition? Just to keep the spec more lean as we may learn new things and it may end up being different and then we would need to modify this spec

Reply:
perfect